vico/multiplatform/src/commonMain/kotlin/com/patrykandpatrick/vico/multiplatform/common/component/ShapeComponent.kt

```diff
@@ -112,14 +137,16 @@ public open class ShapeComponent(
         shape == other.shape &&
         margins == other.margins &&
         strokeFill == other.strokeFill &&
-        strokeThickness == other.strokeThickness
+        strokeThickness == other.strokeThickness &&
+        shadows == other.shadows
 
   override fun hashCode(): Int {
     var result = fill.hashCode()
     result = 31 * result + shape.hashCode()
     result = 31 * result + margins.hashCode()
     result = 31 * result + strokeFill.hashCode()
     result = 31 * result + strokeThickness.hashCode()
+    result = 31 * result + (shadows?.hashCode() ?: 0)
```

Comment:
There’s an `Any?.hashCode` extension function.

```suggestion
    result = 31 * result + shadows.hashCode()
```